### PR TITLE
Fix for issue 132 (pokemon double in list)

### PR DESCRIPTION
**Pull request by Jari27**
Retry. I didn't check style, I can't get it working. :/

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
@Sadye have you tried to simplify the logic in the equals method to:

``` java
 if (o == this) {
    return true;
 } else if (o instanceof CatchablePokemon) {
        return getEncounterId() == ((CatchablePokemon) o).getEncounterId();
 }
return false
```

**Comment by FabianTerhorst** [comment]
travis is failing, because of the missing HashMap import. 

``` java
import java.util.HashMap;
```

**Comment by Jari27** [comment]
Agreed. Fixed both your suggestions.

**Comment by FabianTerhorst** [comment]
You changed the package location with your latest commit and you have to indent the equals method again.

File: src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java [hunk 1]
```diff
@@ -235,12 +266,27 @@ public CatchResult catchPokemon(
...
 		api.getInventories().updateInventories();
 
 		return new CatchResult(response);
 	}
 
+	@Override
+	public boolean equals(Object o) {
+		if (o == this) {
+			return true;
+		} else if (o instanceof CatchablePokemon){
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java]: add a space between ) and {

**Comment by Jari27** [comment]
Thanks for coaching ;)

**Comment by FabianTerhorst** [comment]
looks good to me, travis is building fine. :thumbsup:

**Comment by Grover-c13** [comment]
-   public boolean equals(Object o) {
  -      if (o == this) {
  -          return true;
  -      } else if (o instanceof CatchablePokemon) {
  -          return this.getEncounterId() == ((CatchablePokemon) o)
  -                  .getEncounterId();
  -      }
  -      return false;
  -  }

single letter variables wont conform to the check style sorry, not sure why travis is passing

**Pull request merged**